### PR TITLE
Update empty dots checking

### PR DESCRIPTION
**Pull request by lionel-**
* Rename `check_dots_unused()` to `check_dots_empty()`
* Take `...` and call `nargs()` to simplify implementation and reflect the eagerness of the check.
* Fail rather than warn.
* Export `stop_dots_not_empty()` to throw subclasses of `rlib_error_dots_not_empty`.

### CODE REVIEW

File: R/check.R [hunk 1]
```diff
@@ -85,27 +85,36 @@ check_dots_unnamed <- function(env = parent.frame()) {
...
+#' @inheritParams rlang::abort
+#' @export
+stop_dots_not_empty <- function(message = NULL, .subclass = NULL, ...) {
+  message <- message %||% paste_line(
+    "`...` is not empty.",
+    "",
+    "These dots only exist to allow future extensions and should be empty.",
+    "Did you misspell an argument name?"
   )
+  abort(message, .subclass = c(.subclass, "rlib_error_dots_not_empty"), ...)
```
Comment (hadley) [inline R/check.R]: Why not `ellipsis_error_not_empty`?

Comment (lionel-) [reply R/check.R]: Because this feels like a general error class.

Comment (hadley) [reply R/check.R]: I imagine we'll also have other subclasses for the other check functions — they have a common structure that they're all about dots, and I think they should carry the names of the dots involved as data.

Comment (lionel-) [reply R/check.R]: Agreed, `rlib_error_dots_` reflects this namespace hierarchy: org -> error -> domain

**Pull request merged**